src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -61,7 +61,7 @@ public async Task RunAsync(ActionRunStage stage)
             var compositeGitHubContext = new GitHubContext();
             foreach (var pair in githubContext)
             {
-                compositeGitHubContext[pair.Key] = pair.Value;
+                compositeGitHubContext[pair.Key] = pair.Value.Clone();
```

Comment:
why this will fix the bug? what behavior customer will see when not use composite action at all?

Reply:
Yeah I was wrong closing (sent you a Slack message). 

The issue has to do with: https://github.com/actions/runner/blame/main/src/Runner.Worker/StepsRunner.cs#L109-L116